.github/workflows/ci.yml

```diff
@@ -1,6 +1,11 @@
 name: continuous-integration
 
-on: [push, pull_request]
+on:
+  push:
+    # only pushes to main trigger
+    branches: [main]
+  pull_request:
+    # always triggered
```

Comment:
This change is unrelated to this PR. It stops that the CI is run two time for each PR. I will put this in a separate PR